Add Echo/SQL Sample

petergeorgas opened the pull request:
This branch creates the exact same URL shortening service in the gin-mongo example, but using Echo and PostgreSQL.

Instructions are largely the same as far as running this sample service, with mainly just one additional step of spinning up the PostgreSQL docker container. The container will be configured automatically upon first creation. 

Takes care of #3

slayerjain commented on echo-sql/README.md:
````diff
@@ -0,0 +1,120 @@
+ok      test-app-url-shortener  6.268s  coverage: 80.3% of statements in ./...
+```
+
+**We got 80.3% without writing any testcases or mocks for mongo db!!**
````
we should change this line

slayerjain commented on echo-sql/README.md:
````diff
@@ -0,0 +1,120 @@
+output should look like
+
+```shell
+ok      test-app-url-shortener  6.268s  coverage: 80.3% of statements in ./...
````
We should update this line to reflect the actual code coverage we are getting

petergeorgas replied:
Code coverage has now been updated.

slayerjain commented on echo-sql/README.md:
````diff
@@ -0,0 +1,120 @@
+### Setup URL shortener
+
+```bash
+git clone https://github.com/keploy/example-url-shortener && cd example-url-shortener/echo-sql
````
please update the URL, since it's changed now. We also need to update this for the other gin-mongo example.

petergeorgas replied:
Fixed in most recent commit.

slayerjain reviewed (changes requested):
also, if i shutdown the Postgres database and run tests it fails with the following error: 

<img width="1539" alt="Screenshot 2022-03-15 at 9 33 47 AM" src="https://user-images.githubusercontent.com/12831254/158304998-58f3a82e-0827-415d-9dc3-01baced11d29.png">

@re-Tick let's find out why this is happening.

petergeorgas commented:
@slayerjain Any update on the Postgres shutdown issue? Anything I can do?

slayerjain commented:
@petergeorgas we found another problem with error deserialization. @re-Tick has identified a fix thanks to the amazing go community - https://www.reddit.com/r/golang/comments/tgfvyp/expression_comparing_two_equal_errors_is/

I think we should be able to merge and release within a day :)

slayerjain commented:
@petergeorgas We've released the fix. please update the SDK version to `v0.3.5`and record test cases and run test mode again. Now you should be able to successfully run the test cases even after shutting down Postgres 👯

petergeorgas commented:
@slayerjain All should be good to go. I updated to the most recent Keploy SDK 0.3.5, and updated the README to reflect actual test results. Working really well!

slayerjain commented:
@petergeorgas that's awesome. Checking this out :)

slayerjain commented:
Thanks @petergeorgas  👯
